app/assets/javascripts/posts/editor.js

```diff
@@ -253,29 +253,29 @@ function bindIcon() {
   });
 }
 
-function galleryString(gallery, multiGallery) {
-  var iconsString = "";
+function galleryNode(gallery, multiGallery) {
+  var iconNodes = [];
   var icons = gallery.icons;
 
   for (var i=0; i<icons.length; i++) {
-    iconsString += iconString(icons[i]);
+    iconNodes.push(iconNode(icons[i]));
   }
 
-  if (!multiGallery) return iconsString;
+  if (!multiGallery) return iconNodes;
 
-  var nameString = "<div class='gallery-name'>" + gallery.name + "</div>";
-  return "<div class='gallery-group'>" + nameString + iconsString + "</div>";
+  var nameNode = $("<div class='gallery-name'></div>").text(gallery.name);
+  return $("<div class='gallery-group'>").append(nameNode).append(iconNodes);
 }
 
-function iconString(icon) {
+function iconNode(icon) {
   var imgId = icon.id;
   var imgUrl = icon.url;
   var imgKey = icon.keyword;
   shownIcons.push(icon.id);
 
-  if (!icon.skip_dropdown) $("#icon_dropdown").append($("<option>").attr({value: imgId}).append(imgKey));
+  if (!icon.skip_dropdown) $("#icon_dropdown").append($("<option>").attr({value: imgId}).text(imgKey));
   var iconImg = $("<img>").attr({src: imgUrl, alt: imgKey, title: imgKey, 'class': 'icon img-'+imgId, 'data-icon-id': imgId});
-  return $("<div>").attr('class', 'gallery-icon').append(iconImg).append("<br />").append(imgKey)[0].outerHTML;
+  return $("<div>").attr('class', 'gallery-icon').append(iconImg).append("<br />").append(document.createTextNode(imgKey));
```

Comment:
Do not totally understand what's up with the text node thing here

Reply:
We can't use `.text`, because that overrides all of the contents - we just want to add text to it. `.text` basically wraps the text in a DOM text node, which makes sure it isn't interpreted as HTML, but instead as text - this should let us append text safely here